common/Util-desktop.cpp

```diff
@@ -160,6 +162,43 @@ std::size_t getTotalSystemMemoryKb()
     return totalMemKb;
 }
 
+size_t getMaxConcurrentTCPConnections(const size_t cutOffMinimum)
+{
+#ifdef __linux__
+    auto readDecimal = [](const char* path, const size_t defaultValue) -> size_t {
+        FILE* file = fopen(path, "r");
+        if (file == nullptr)
+            return defaultValue;
+        char line[4096] = { 0 };
+        // coverity[tainted_data_argument : FALSE] - we trust the kernel-provided data
+        const char *line2 = fgets(line, sizeof(line), file);
+        fclose(file);
+        if(line2 == nullptr)
+            return defaultValue;
+        char* endptr = nullptr;
+        errno = 0;  // Flush previous error indicator. Reminder: errno is thread-local
+        const unsigned long long num = std::strtoull(line, &endptr, 10);
+        if (0 != errno || nullptr == endptr || endptr == line || num > std::numeric_limits<size_t>::max())
+            return defaultValue;
+        return size_t(num);
+    };
+    // - 4 tcp_max_orphans sockets per MB
+    // - {4096M -> 16384}, {16384M -> 65536}, {65407M -> 262144}, ...
+    const ssize_t tcp_max_orphans = readDecimal("/proc/sys/net/ipv4/tcp_max_orphans", 0); // ignored if n/a
+    const ssize_t nf_conntrack_max = readDecimal("/proc/sys/net/nf_conntrack_max", 0); // ignored if n/a
+    const size_t res_min = nf_conntrack_max > 0 ? std::min(tcp_max_orphans, nf_conntrack_max) : tcp_max_orphans;
+    const size_t res_clipped = res_min >= cutOffMinimum ? res_min : 0;
+    LOG_DBG("MaxConcurrentTCPConnections: clip(min(orphans " << tcp_max_orphans
+            << ", conntrack " << nf_conntrack_max << ") -> "
+            << res_min
+            << ", cut-off-min " << cutOffMinimum
+            << ") -> " << res_clipped);
+    return res_clipped;
```

Comment:
Seems it is very likely that  this function returns zero - is that expected ?

The cutOffMinimum and the way it is injected here is really non-intuitive. Can you add a one-line comments against each line of your ? operator logic above to explain what is going on. We have two metaphores for the same thing 'cut-off' and 'clipped' - please choose something consistent for the variable naming to give the next poor reader a clue.

Reply:
Moved logic to COOLWSD, simplified getter.